app/domain/SlpFileWriter.js

```diff
@@ -37,6 +46,59 @@ export default class SlpFileWriter {
 
   updateSettings(settings) {
     this.folderPath = settings.targetFolder;
+    this.obsIP = settings.obsIP;
+    this.obsSourceName = settings.obsSourceName;
+  }
+
+  async connectOBS() {
+    if (this.obsIP && this.obsSourceName) {
+      await this.obs.connect({address: this.obsIP});
+      this.scenes = await this.obs.send("GetSceneList");
+      this.scenes = this.scenes.scenes;
+    }
+  }
+
+  disconnectOBS() {
+    this.obs.disconnect();
+  }
+
+  setStatus(value) {
+    this.statusOutput.status = value;
+    console.log(`Status changed: ${value}`);
+    for (let i = this.scenes.length - 1; i >= 0; i -= 1) {
```

Comment:
what happens if scenes was never set? Also I tend to avoid for loops in javascript and tend to prefer lodash instead.

```
const scenes = this.scenes || [];
const pairs = _.flatMap(scenes, (scene) => {
  const sources = scene.sources || [];
  return _.map(sources, (source) => { scene: scene, source: source };
});
_.forEach(pairs, (pair) => {
  const res = this.obs.send(
    "SetSceneItemProperties", 
    {"scene-name": this.scenes[j].name, "item": this.obsSourceName, "visible": value}
  );
  console.log(res);
});
```